internal/providers/terraform/azure/key_vault_key.go

```diff
@@ -0,0 +1,170 @@
+		return nil
+	}
+
+	skuName = strings.Title(keyVault[0].Get("sku_name").String())
+	keyType = d.Get("key_type").String()
+
+	if d.Get("key_size").Type != gjson.Null {
+		keySize = d.Get("key_size").String()
+	}
+
+	var costComponents []*schema.CostComponent
+
+	unit := "10K transactions"
+
+	var secretsTransactions *decimal.Decimal
+	if u != nil && u.Get("monthly_secrets_operations").Exists() {
+		secretsTransactions = decimalPtr(decimal.NewFromInt(u.Get("monthly_secrets_operations").Int()))
+	}
+	meterName = "Operations"
+	costComponents = append(costComponents, vaultKeysCostComponent("Secrets operations", location, unit, skuName, meterName, "0", secretsTransactions, 10000))
+
+	var certificateRenewals, certificateOperations *decimal.Decimal
+	if u != nil && u.Get("monthly_certificate_renewal_requests").Exists() {
+		certificateRenewals = decimalPtr(decimal.NewFromInt(u.Get("monthly_certificate_renewal_requests").Int()))
+	}
+	meterName = "Certificate Renewal Request"
+	costComponents = append(costComponents, vaultKeysCostComponent("Certificate operations", location, "renewals", skuName, meterName, "0", certificateRenewals, 1))
+
+	if u != nil && u.Get("monthly_certificate_all_other_operations").Exists() {
+		certificateOperations = decimalPtr(decimal.NewFromInt(u.Get("monthly_certificate_all_other_operations").Int()))
+	}
+	meterName = "Operations"
+	costComponents = append(costComponents, vaultKeysCostComponent("Certificate operations", location, unit, skuName, meterName, "0", certificateOperations, 10000))
+
+	var keyRotationRenewals *decimal.Decimal
+	if u != nil && u.Get("monthly_key_rotation_renewals").Exists() {
+		keyRotationRenewals = decimalPtr(decimal.NewFromInt(u.Get("monthly_key_rotation_renewals").Int()))
+	}
+	meterName = "Secret Renewal"
+	costComponents = append(costComponents, vaultKeysCostComponent("Storage key rotation", location, "renewals", skuName, meterName, "0", keyRotationRenewals, 1))
+
+	if !strings.HasSuffix(keyType, "HSM") {
+		var softwareProtectedTransactions *decimal.Decimal
+		if u != nil && u.Get("monthly_protected_keys_operations").Exists() {
+			softwareProtectedTransactions = decimalPtr(decimal.NewFromInt(u.Get("monthly_protected_keys_operations").Int()))
+		}
+
+		name := "Software-protected keys"
+		meterName = "Operations"
+
+		if keyType == "RSA" && keySize == "2048" {
+			costComponents = append(costComponents, vaultKeysCostComponent(name, location, unit, skuName, meterName, "0", softwareProtectedTransactions, 10000))
+		} else {
+			meterName = "Advanced Key Operations"
+			costComponents = append(costComponents, vaultKeysCostComponent(name, location, unit, skuName, meterName, "0", softwareProtectedTransactions, 10000))
+		}
+	}
+
+	if strings.HasSuffix(keyType, "HSM") && skuName == "Premium" {
+		var protectedKeys, hsmProtectedTransactions *decimal.Decimal
+
+		name := "HSM-protected keys"
+		keyUnit := "Per key per month"
```

Comment:
I think this can just be "months", in the same way we have "hours" for instances. We did the same for aws_kms_key and google_kms_crypto_key